go/processor/floria/processor_test.go

```diff
@@ -0,0 +1,89 @@
+// Copyright (c) 2024 Fantom Foundation
+//
+// Use of this software is governed by the Business Source License included
+// in the LICENSE file and at fantom.foundation/bsl11.
+//
+// Change Date: 2028-4-16
+//
+// On the date above, in accordance with the Business Source License, use of
+// this software will be governed by the GNU Lesser General Public License v3.
+
+package floria
+
+import (
+	"testing"
+
+	test_processor "github.com/Fantom-foundation/Tosca/go/integration_test/processor"
+	"github.com/Fantom-foundation/Tosca/go/tosca"
+)
+
+func TestProcessorRegistry_InitProcessor(t *testing.T) {
+	processorFactories := tosca.GetAllRegisteredProcessorFactories()
+	if len(processorFactories) == 0 {
+		t.Errorf("No processor factories found")
+	}
+
+	processor := tosca.GetProcessorFactory("floria")
+	if processor == nil {
+		t.Errorf("Floria processor factory not found")
+	}
+}
+
+func TestProcessor_HandleNonce(t *testing.T) {
+	transaction := tosca.Transaction{
+		Sender: tosca.Address{1},
+		Nonce:  9,
+	}
+
+	context := test_processor.NewScenarioContext()
```

Comment:
I would argue that at the scope of unit testing, mocks for the `TransactionContext` may be more adequate. They provide a more fine-grained control on the expected effects and simplify the testing of the post-condition.
```
        ctrl := gomock.NewController(t)
	context := tosca.NewMockTransactionContext(ctrl)

	context.EXPECT().GetNonce(tosca.Address{1}).Return(uint64(9))
	context.EXPECT().SetNonce(tosca.Address{1}, uint64(10))

	transaction := tosca.Transaction{
		Sender: tosca.Address{1},
		Nonce:  9,
	}

	err := handleNonce(transaction, context)
	if err != nil {
		t.Errorf("handleNonce returned an error: %v", err)
        }
```

Reply:
Right, changed it to mocks for all unit tests.